File: fathersDay/Dsa/Day 28 (796. Rotate String)/Solution.java
class Solution {
    public boolean RotateString(String s, String goal) {
        if (s.length() != goal.length()) {
            return false;
        }
        String doubledS = s + s;
        return doubledS.contains(goal);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String s = "abcde";
        String goal = "cdeab";
        boolean result = solution.RotateString(s, goal);
        System.out.println(result); // Output: true
    }
}

File: fathersDay/Dsa/Day 28 (796. Rotate String)/Solution.cpp
#include <iostream>
#include <string>

class Solution {
public:
    bool RotateString(std::string s, std::string goal) {
        if (s.length() != goal.length()) {
            return false;
        }
        std::string doubled_s = s + s;
        return doubled_s.find(goal) != std::string::npos;
    }
};

int main() {
    Solution solution;
    std::string s = "abcde";
    std::string goal = "cdeab";
    bool result = solution.RotateString(s, goal);
    std::cout << (result ? "True" : "False") << std::endl; // Output: True
    return 0;
}
